src/itdelatrisu/opsu/beatmap/BeatmapParser.java

```diff
@@ -732,12 +734,21 @@ public static void parseHitObjects(Beatmap beatmap) {
 					hitObject.setComboIndex(comboIndex);
 					hitObject.setComboNumber(comboNumber++);
 
-					beatmap.objects[objectIndex++] = hitObject;
+					objects.add(hitObject);
+					objectIndex++;
 				} catch (Exception e) {
 					Log.warn(String.format("Failed to read hit object '%s' for beatmap '%s'.",
 							line, beatmap.toString()), e);
 				}
 			}
+			objects.trimToSize();
```

Comment:
There's no point calling this method for local variables. (It might even result in more space getting allocated before GC.)